detox/android/detox/build.gradle

```diff
@@ -80,18 +80,18 @@ dependencies {
 }
 
 // androidx.test deps.
-// All are aligned with this release: https://developer.android.com/jetpack/androidx/releases/test#1.2.0
+// All are aligned with this release: https://developer.android.com/jetpack/androidx/releases/test#1.4.0
 dependencies {
 
-    // Versions are in-sync with the 'androidx-test-1.2.0' release/tag of the android-test github repo,
-    // used by the Detox generator. See https://github.com/android/android-test/releases/tag/androidx-test-1.2.0
+    // Versions are in-sync with the 'androidx-test-1.4.0' release/tag of the android-test github repo,
+    // used by the Detox generator. See https://github.com/android/android-test/releases/tag/androidx-test-1.4.0
     // Important: Should remain so when generator tag is replaced!
-    api('androidx.test.espresso:espresso-core:3.3.0') { // Needed all across Detox but also makes Espresso seamlessly provided to Detox users with hybrid apps/E2E-tests.
+    api('androidx.test.espresso:espresso-core:3.4.0') { // Needed all across Detox but also makes Espresso seamlessly provided to Detox users with hybrid apps/E2E-tests.
         exclude group: 'com.google.code.findbugs', module: 'jsr305'
     }
-    api 'androidx.test.espresso:espresso-web:3.3.0' // Web-View testing
-    api 'androidx.test:rules:1.2.0' // Needed because of ActivityTestRule. Needed by users *and* internally used by Detox.
-    api 'androidx.test.ext:junit:1.1.1' // Needed so as to seamlessly provide AndroidJUnit4 to Detox users. Depends on junit core.
+    api 'androidx.test.espresso:espresso-web:3.4.0' // Web-View testing
+    api 'androidx.test:rules:1.4.0' // Needed because of ActivityTestRule. Needed by users *and* internally used by Detox.
+    api 'androidx.test.ext:junit:1.1.3' // Needed so as to seamlessly provide AndroidJUnit4 to Detox users. Depends on junit core.
```

Comment:
Consider boy-scouting by replacing the comments with a `because` (example in line 104 below - https://github.com/wix/Detox/pull/3035/files#diff-ef16074b7489954843e57dfef5376da6115f35ea0b6be19781edcb542530437bR104)